### PR TITLE
replace match group script pack

### PR DESCRIPTION
**Pull request by guyfreund**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/22102

## Description
New transformer to replace all group matches of regex pattern.

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [ ] Code Review

## Technical Writer Review
- [ ] [YML](https://github.com/demisto/content/blob/73b0389393a8b125e863962493113027daeef124/Packs/ReplaceMatchGroup/Scripts/ReplaceMatchGroup/ReplaceMatchGroup.yml)

### CODE REVIEW

**Comment by guyfreund** [comment]
@David-BMS FYI.

File: Packs/ReplaceMatchGroup/Scripts/ReplaceMatchGroup/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1 @@
+## [Unreleased]
```
Comment (Itay4) [inline Packs/ReplaceMatchGroup/Scripts/ReplaceMatchGroup/CHANGELOG.md]: remove this

File: Packs/ReplaceMatchGroup/Scripts/ReplaceMatchGroup/ReplaceMatchGroup.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+comment: Returns a string with all matches of a regex pattern groups replaced by a replacement
+enabled: true
+args:
+- name: value
+  required: true
+  description: An array of email addresses to be filtered by domain.
+- name: regex
+  required: true
+  description: A regex pattern who's groups to be replaced by the replaceWith argument
+- name: replaceWith
```
Comment (Itay4) [inline Packs/ReplaceMatchGroup/Scripts/ReplaceMatchGroup/ReplaceMatchGroup.yml]:
```suggestion
- name: replace_with
```

**Comment by David-BMS** [comment]
@guyfreund I'm seeing some unexpected behavior with the script please dont merge until we review it together

File: Packs/ReplaceMatchGroup/Scripts/ReplaceMatchGroup/ReplaceMatchGroup.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+            end = match.start(index + 1)
+            output.append(value[start:end])
+            output.append(replace_with)
+            start = match.end(index + 1)
+    output.append(value[start:])  # Handling the tail of the string
+
+    demisto.results(''.join(output))
+
+
+if __name__ == "__builtin__" or __name__ == "builtins" or __name__ == "__main__":
```
Comment (Itay4) [inline Packs/ReplaceMatchGroup/Scripts/ReplaceMatchGroup/ReplaceMatchGroup.py]: check if in tuple instead

File: Packs/ReplaceMatchGroup/CHANGELOG.md [hunk 4]
```diff
@@ -0,0 +1 @@
+## [Unreleased]
```
Comment (Itay4) [inline Packs/ReplaceMatchGroup/CHANGELOG.md]: remove changelog

File: Packs/ReplaceMatchGroup/Scripts/ReplaceMatchGroup/ReplaceMatchGroup.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+    for match in regex.finditer(value):
+        for index, _ in enumerate(match.groups()):
+            end = match.start(index + 1)
+            output.append(value[start:end])
+            output.append(replace_with)
+            start = match.end(index + 1)
+    output.append(value[start:])  # Handling the tail of the string
+
+    demisto.results(''.join(output))
```
Comment (Itay4) [inline Packs/ReplaceMatchGroup/Scripts/ReplaceMatchGroup/ReplaceMatchGroup.py]: return a value here and call `results` in the main block

File: Packs/ReplaceMatchGroup/Scripts/ReplaceMatchGroup/ReplaceMatchGroup_test.py [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    ('{"1"},{"2"},{"3"},{"4"}', '\{".*?"\}(,)\{".*?"\}', ':', '{"1"}:{"2"},{"3"}:{"4"}'),
+    ('this is a test, this is another test.', '.*?(,).*?(\.)', '!', 'this is a test! this is another test!'),
+    ('a, b. c,.', '.*?(,|\.)', '!', 'a! b! c!!'),
+    ('.this is a test, this is another test.', '(\.).*?(,).*?(\.)', '!', '!this is a test! this is another test!'),
+    ('[{"UrlValue:https://sslmanageamazones.ddns.net/info.ppl.service.access/paypal"}, '
+     '{"UrlValue:https://sslmanageamazones.ddns.net/info.ppl.service.access/paypal"}, '
+     '{"UrlValue:https://sslmanageamazones.ddns.net/info.ppl.service.access/paypal"}]', '.*?(:).*?:\\/\\/', '":"',
+     '[{"UrlValue":"https://sslmanageamazones.ddns.net/info.ppl.service.access/paypal"}, '
+     '{"UrlValue":"https://sslmanageamazones.ddns.net/info.ppl.service.access/paypal"}, '
+     '{"UrlValue":"https://sslmanageamazones.ddns.net/info.ppl.service.access/paypal"}]')
```
Comment (Itay4) [inline Packs/ReplaceMatchGroup/Scripts/ReplaceMatchGroup/ReplaceMatchGroup_test.py]: add a test case in which no matches are found and verify string is returned as the value given

File: Packs/ReplaceMatchGroup/Scripts/ReplaceMatchGroup/ReplaceMatchGroup.py [hunk 7]
```diff
@@ -0,0 +1,24 @@
+import demistomock as demisto
+import re
+
+
+def main():
+    value = demisto.args()['value']
```
Comment (Itay4) [inline Packs/ReplaceMatchGroup/Scripts/ReplaceMatchGroup/ReplaceMatchGroup.py]: pass `demisto.args()` to the function so you won't have to call it 3 times

**Comment by guyfreund** [comment]
@kirbles19 - as @David-BMS talked to you on Sunday, the yml file is ready to be reviewed.

**Comment by guyfreund** [comment]
> remove changelog and readme

deleted changelog, readme cannot be deleted due to pack hooks.

**Comment by guyfreund** [comment]
@kirbles19 an ETA here?

**Comment by David-BMS** [comment]
@kirbles19 This is becoming very critical for the GA and some TIM playbooks please approve

**Comment by kirbles19** [comment]
@guyfreund Done.

**Pull request closed without merging**